server/controllers/connectedAccounts.js

```diff
@@ -37,11 +43,81 @@ module.exports = (app) => {
     get: (req, res, next) => {
       const payload = req.jwtPayload;
       const provider = req.params.service;
-      if (payload.scope === provider && payload.username) {
+      if (payload.scope === 'connected-account' && payload.username) {
         res.send({provider, username: payload.username, connectedAccountId: payload.connectedAccountId})
       } else {
         return next(new errors.BadRequest('Github authorization failed'));
       }
+    },
+
+    createGroupFromGithubRepo: (req, res, next) => {
+      var payload = req.required.payload;
+
+      const connectedAccountId = req.jwtPayload.connectedAccountId;
+
+      var creator;
+      const group = payload.group;
+      const contributors = payload.users;
+      const creatorGithubUsername = payload.github_username;
+
+      ConnectedAccount
+        .findOne({
+          where: { id: connectedAccountId },
+          include: { model: User }
+        })
+        .then(ca => creator = ca.User)
+        .then(() => Group.create(group))
+        .then(group => {
+
+          const options = {
+            role: roles.MEMBER,
+            remoteUser: creator
+          };
+          async.auto({
+            createActivity: (cb) => {
+              Activity.create({
+                type: 'group.created',
+                UserId: creator.id,
+                GroupId: group.id,
+                data: {
+                  group: group.info,
+                  user: creator.info
+                }
+              }).done(cb);
+            },
+
+            addCreator: ['createActivity', (cb) => {
+              groups._addUserToGroup(group, creator, options, cb);
+            }],
+
+            addContributors: ['addCreator', (cb) => {
+              // TODO: find a cleaner way of doing this
+              async.each(contributors, (user, callback) => {
+                // since we added the creator above with an email, avoid double adding
+                if (user !== creatorGithubUsername) {
+                  var newUser = {
+                    name: `${user}.github`,
+                    email: `${user}.github@opencollective.com`, // TODO: see if we can get their real email
```

Comment:
`email` field in the `Users` table can't be null. That's the main reason for stuffing an email address.

We could remove that constraint - i had a feeling it'll cause other issues throughout the codebase, since we expect `email` field to be present for any user.

Creating a ConnectedAccount and storing the github info there makes a lot of sense. I'll add that.